dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-voice/src/main/java/org/apache/dolphinscheduler/plugin/alert/voice/VoiceAlertChannel.java

```diff
@@ -0,0 +1,84 @@
+ */
+
+package org.apache.dolphinscheduler.plugin.alert.voice;
+
+import static org.apache.dolphinscheduler.plugin.alert.voice.VoiceAlertConstants.NAME_ACCESS_KEY_ID;
+import static org.apache.dolphinscheduler.plugin.alert.voice.VoiceAlertConstants.NAME_ACCESS_KEY_SECRET;
+import static org.apache.dolphinscheduler.plugin.alert.voice.VoiceAlertConstants.NAME_ADDRESS;
+import static org.apache.dolphinscheduler.plugin.alert.voice.VoiceAlertConstants.NAME_CALLED_NUMBER;
+import static org.apache.dolphinscheduler.plugin.alert.voice.VoiceAlertConstants.NAME_CALLED_SHOW_NUMBER;
+import static org.apache.dolphinscheduler.plugin.alert.voice.VoiceAlertConstants.NAME_TTS_CODE;
+
+import org.apache.dolphinscheduler.alert.api.AlertChannel;
+import org.apache.dolphinscheduler.alert.api.AlertData;
+import org.apache.dolphinscheduler.alert.api.AlertInfo;
+import org.apache.dolphinscheduler.alert.api.AlertResult;
+
+import java.util.Map;
+
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public final class VoiceAlertChannel implements AlertChannel {
+
+    @Override
+    public AlertResult process(AlertInfo info) {
+
+        AlertData alert = info.getAlertData();
+        Map<String, String> paramsMap = info.getAlertParams();
+        if (null == paramsMap) {
+            return new AlertResult("false", "aliyun-voice params is null");
+        }
+        VoiceParam voiceParam = buildVoiceParam(paramsMap);
+        VoiceSender voiceSender = new VoiceSender(voiceParam);
+
+        AlertResult alertResult = voiceSender.send();
+        Boolean flag = Boolean.parseBoolean(String.valueOf(alertResult.getStatus()));
+        if (flag) {
+            log.info("alert send success");
+            alertResult.setStatus("true");
+            alertResult.setMessage("vocie send success.");
+        } else {
+            alertResult.setMessage("vocie send error.");
+            log.info("alert send error : {}", alertResult.getMessage());
+        }
+
+        return alertResult;
+    }
+
+    public VoiceParam buildVoiceParam(Map<String, String> paramsMap) {
+        String calledNumber = paramsMap.get(NAME_CALLED_NUMBER);
+        String calledShowNumber = paramsMap.get(NAME_CALLED_SHOW_NUMBER);
+        String ttsCode = paramsMap.get(NAME_TTS_CODE);
+        VoiceParam param = new VoiceParam();
+        param.setCalledNumber(calledNumber);
+        param.setCalledShowNumber(calledShowNumber);
+        param.setTtsCode(ttsCode);
+        VoiceParam.Connection connection = new VoiceParam.Connection();
+        String address = paramsMap.get(NAME_ADDRESS);
+        String accessKeyId = paramsMap.get(NAME_ACCESS_KEY_ID);
+        String accessKeySecret = paramsMap.get(NAME_ACCESS_KEY_SECRET);
+        connection.setAddress(address);
+        connection.setAccessKeyId(accessKeyId);
+        connection.setAccessKeySecret(accessKeySecret);
+        param.setConnection(connection);
+        String outId = String.valueOf((int) (1000 + Math.random() * (99999999 - 1 - 10000000)));
```

Comment:
The outId is strange, is there doc about this param?

Reply:
callback , no practical significance , the ID reserved for the caller to use，